python/letsql/common/utils/dask_normalize_expr.py

```diff
@@ -167,6 +171,44 @@ def normalize_module(module):
     )
 
 
+@dask.base.normalize_token.register(Read)
+def normalize_read(read):
+    path = dict(read.read_kwargs).get("path") or dict(read.read_kwargs).get("source")
+    if isinstance(path, (str, pathlib.Path)):
+        path = str(path)
+        if path.startswith("http") or path.startswith("https:"):
+            import requests
+
+            resp = requests.head(path)
+            resp.raise_for_status()
+            dct = {
```

Comment:
Is the variable `dct` a dictionary?

Reply:
yes

Reply:
rather, it should be (thought doesn't need to be)